sixtyfps_runtime/corelib/properties.rs

```diff
@@ -2387,10 +2387,16 @@ pub(crate) mod ffi {
             }
         }
 
+        impl CStateBinding {
+            fn call(&self) -> i32 {
+                (self.binding)(self.user_data)
+            }
+        }
+
         let c_state_binding = CStateBinding { binding, user_data, drop_user_data };
         let bind_callable = StateInfoBinding {
             dirty_time: Cell::new(None),
-            binding: move || (c_state_binding.binding)(c_state_binding.user_data),
+            binding: move || c_state_binding.call(),
```

Comment:
Ohh, so here it "moved" just the two fields (`binding` and `user_data`) and then called `drop` on the `c_state`, which called the `x` "destructor"?

Reply:
yes.